### PR TITLE
Add support for customized sources URL option

### LINKED ISSUE #1124: Parameterize openjdk sources used for testing
This is somewhat related to https://github.com/AdoptOpenJDK/openjdk-tests/issues/1082

We'd like to automatically run the test pipeline on upstream OpenJDK hotspot Early Access (EA) builds so as to be able to automate consistent testing across platform specific builds and have test results available publicly.

While [get.sh](https://github.com/AdoptOpenJDK/openjdk-tests/blob/master/get.sh) has parameters for the JDK under test via `--sdk_resource customized` and `--customizedURL <url>`, the same is not possible for matching JDK sources. It is assumed the sources are in "sync" at AdoptOpenJDK/openjdk-jdkXu, where `X` is the JDK version like 8, 11, 12, etc. This assumption doesn't hold for EA upstream builds which have been built from sources *ahead* of the AdoptOpenJDK git mirrors. This is mostly an issue of running OpenJDK regression tests on customized builds.

In order to make this more flexible, I'd propose to implement something similar to `--sdk_resource` and `--customizedURL <url>`, but for JDK sources. Once that's implemented, Jenkins files could get enhanced to expose those new parameters. Then it should be possible to run the test pipeline on upstream builds automatically as soon as they become available at:
https://github.com/AdoptOpenJDK/openjdk8-upstream-binaries/releases and
https://github.com/AdoptOpenJDK/openjdk11-upstream-binaries/releases

### PR DESCRIPTION
**Pull request by jerboaa**
Closes #1124

Signed-off-by: Severin Gehwolf <sgehwolf@redhat.com>

### CODE REVIEW

**Comment by jerboaa** [comment]
Works for me for local testing and verified via Grinder_Sandbox:
https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder_Sandbox/383/

Please review!

**Comment by llxia** [comment]
Thanks @jerboaa 

Just a thought, `CUSTOMIZED_SDK_URL` does take multiple URLs (separated with space). Instead of creating a new parameter `CUSTOMIZED_SDK_SOURCE_URL `, it is possible to leverage the existing parameter `CUSTOMIZED_SDK_URL`?

File: get.sh [hunk 1]
```diff
@@ -322,6 +349,9 @@ parseCommandLineArgs "$@"
 if [[ "$SDKDIR" != "" ]]; then
 	getBinaryOpenjdk
 fi
+if [[ "$TESTDIR" != "" ]]; then
+	getOpenJDKSources
+fi
```
Comment (sophia-guo) [inline get.sh]: $TESTDIR should never be empty. I would suggest moving the condition `if [ "$SDK_RESOURCE" == "customized" ] && [ "$CUSTOMIZED_SDK_SOURCE_URL" != "" ]` out of function `getOpenJDKSources()` as

```
if [ "$SDK_RESOURCE" == "customized" ] && [ "$CUSTOMIZED_SDK_SOURCE_URL" != "" ]; then
         getOpenJDKSources
fi

**Comment by jerboaa** [comment]
> Thanks @jerboaa
> 
> Just a thought, `CUSTOMIZED_SDK_URL` does take multiple URLs (separated with space). Instead of creating a new parameter `CUSTOMIZED_SDK_SOURCE_URL `, it is possible to leverage the existing parameter `CUSTOMIZED_SDK_URL`?

Perhaps. It feels wrong to "abuse" a parameter which is called "CUSTOMIZED_SDK_URL", though. That logic already does too much for my taste. To me a new parameter is cleaner.

File: get.sh (hunk 1 above)
Comment (jerboaa) [reply get.sh]: Thanks, fixed.

**Comment by jerboaa** [comment]
I see a merge button. So not sure how you handle PRs being merged in terms of workflow. Is the PR submitter supposed to merge, once approved by a reviewer? Do you use a different workflow? Please let me know.

**Comment by jerboaa** [comment]
Also: Thanks for the review, Sophia!

**Comment by smlambert** [comment]
Let's proceed with the approach that the reviewer merges the PR.  We have been 'loose' with this as an approach to date.

This is barring the exceptional case that a change needs to be co-ordinated with another PR (potentially in another repo).  In the exceptional case, it should be stated clearly in the PR, so that it can be clearly communicated and co-ordinated.

@sophia-guo - if you are happy with @jerboaa 's changes, please merge them (and his other PR when ready).

**Comment by jerboaa** [comment]
Thanks all!

**Pull request merged**